src/app/person/person.component.ts

```diff
@@ -9,38 +9,45 @@ import { PersonService } from "src/app/services/person.component.service";
   styleUrls: ["./person.component.scss"]
 })
 export class PersonComponent implements OnInit {
-  constructor(public router: Router, private PersonService: PersonService) {}
+  personList: Array<object> = this.personService
+    .getPersons()
+    .sort((a: any, b: any) => a.id - b.id);
+  constructor(public router: Router, private personService: PersonService) {}
 
   ngOnInit() {}
 
-  return() {
+  navigateBack() {
     this.router.navigate(["todo"]);
   }
 
   //--------- CRUD Operations -----------
 
   addPersonFromService() {
-    this.PersonService.addPerson({
-      id: this.personList.length + 1,
+    this.personService.addPerson({
+      id: this.personService.nextId,
       name: "hagay"
     });
+    this.personService.getPersons().sort((a: any, b: any) => a.id - b.id);
   }
 
   updatePersonFromService() {
-    this.PersonService.updatePerson(3, "hagay");
+    this.personService.updatePerson(3, "hagay");
   }
 
   deletePersonFromService() {
-    this.PersonService.deletePerson(3);
+    this.personService
+      .deletePerson(3)
+      .then(
+        () =>
+          (this.personList = this.personService
+            .getPersons()
+            .sort((a: any, b: any) => a.id - b.id))
+      );
   }
 
   getAllPersonsFromService() {
-    console.log(this.PersonService.getPersons());
+    this.personService.getPersons();
```

Comment:
You are doing here nothing. 
You don't return a value and you do not do anything with the person's info. 
I